lib/drivers/medtronic600/medtronic600Simulator.js

```diff
@@ -506,13 +506,21 @@ class Medtronic600Simulator {
         }
       } else {
         // If a suspended basal exists when there's no activeBasal
-        // (ie, first event is a suspend), modify it
+        // (ie, first event is a suspend), modify it, if it overlaps
+        // a basal schedule.
         if (activeSuspend && _.isNull(activeBasal)) {
           activeSuspend.time = basal.time;
           activeSuspend.deviceTime = basal.deviceTime;
           activeSuspend.duration = Date.parse(activeSuspend.basalEndTime) - Date.parse(basal.time);
           Medtronic600Simulator.setSuppressedBasal(activeSuspend, basal);
-          Medtronic600Simulator.checkDuration(activeSuspend);
+
+          if (activeSuspend.duration <= 0) {
+            // If the suspend duration is not a positive number, then the suspend
+            // is not overlapping a scheduled basal.
+            // Delete the suspend, since we don't know what it's suppressing.
+            this.removeEvent(activeSuspend);
```

Comment:
Maybe just don't add suppressed info if it's not available, instead of removing the suspend event?

Reply:
See above comment